Update docs for Declarative Automation (#24672)
dagster-io/dagster: docs/content/concepts/automation/declarative-automation/customizing-automation-conditions.mdx

  </tbody>
</table>

### Composite conditions

Finally, there are a set of pre-built conditions which make it easier to construct common combinations of the above conditions.

| Condition                                         | Description                                                                                            |
| ------------------------------------------------- | ------------------------------------------------------------------------------------------------------ |

> | `AutomationCondition.any_deps_updated`            | Returns true if any dependencies have been updated since the previous tick                             |

Reviewer: Two comments
- For all the ones below, I would avoid `Returns true` language and instead focus on the outcome. So maybe `Executes if ...` or `Starts a run if ...` etc. 
- Avoiding the tick language if possible would be nice here too. Maybe something like `Materializes this asset if any of its dependencies have been updated since the last scheduled update`

Author: I avoided the "starts a run if" language because these conditions are generally not recommended as standalone policies to put on an asset. i.e. you really would not want to have `@asset(automation_condition=AutomationCondition.any_deps_missing())` (why would you want to materialize if any dependencies were missing?)

Maybe just removing the "returns true if" entirely is a bit clearer overall.

Would swapping "tick" to "evaluation" feel a bit better? In this case, it is fairly relevant that the value returned by these conditions is dependent on things that happen between discrete evaluations. Made that update.

Reviewer: Ah that makes sense, yea these are composable. I think The update looks good!

| `AutomationCondition.any_deps_missing`            | Returns true if any dependencies have never been materialized or observed                              |
| `AutomationCondition.any_deps_in_progress`        | Returns true if any dependencies are part of an in-progress run                                        |
| `AutomationCondition.all_deps_updated_since_cron` | Returns true if all dependencies have been updated since the latest tick of the provided cron schedule |

---

## Modifying policies

> It's common to have use cases which are similar to pre-built policies, but with minor differences. While it is always possible to simply copy the base implementation and make modifications to it as needed, it can often be simpler to use the `.without()` method to remove the unwanted sub-conditions, or add additional conditions with the `&` operator.

Reviewer: ```suggestion
It's common to have use cases similar to pre-built policies but with minor differences. While it is always possible to copy the base implementation and modify it as needed, it can often be simpler to use the `.without()` method to remove the unwanted sub-conditions or add additional conditions with the `&` operator.
```

### `AutomationCondition.eager()`: Ignoring missing dependencies

> By default, `AutomationCondition.eager()` will not materialize an asset partition if it has any missing dependencies. However, in some cases it is expected to have missing upstream data, and this should not prevent execution from occurring. This can be accomplished by removing `~AutomationCondition.any_deps_missing()` from the eager policy:

Reviewer: ```suggestion
By default, `AutomationCondition.eager()` will not materialize an asset partition if it has any missing dependencies. If it is expected to have missing upstream data, remove `~AutomationCondition.any_deps_missing()` from the eager policy to allow execution:
```
